##### import 

In [53]:
from sklearn.ensemble import RandomForestClassifier
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score
import pandas as pd

##### load data

In [54]:
all_var_df = pd.read_csv('../data/clean_data_model.csv')
selected_feat_df = pd.read_csv('../data/data_model.csv')

#### Creating the feature and target variables from the above read data

In [55]:
X_all = all_var_df.drop(["diagnosis"], axis=1)
y_all = all_var_df["diagnosis"]
y_selected_feat_df = selected_feat_df.drop(["diagnosis"], axis=1)
x_selected_feat_df = selected_feat_df["diagnosis"]

In [56]:
# Train test split for the feature extracted dataframe

X_train_all, X_test_all, y_train_all, y_test_all = train_test_split(
    X_all, y_all,test_size=0.2, random_state=10
)

In [57]:

X_train_feat, X_test_feat, y_train_feat, y_test_feat = train_test_split(
    y_selected_feat_df, x_selected_feat_df,test_size=0.2, random_state=10
)

In [58]:
classifier_all = RandomForestClassifier(n_estimators=100)
classifier_all.fit(X_train_all, y_train_all)

RandomForestClassifier()

In [59]:

classifier_feature= RandomForestClassifier(n_estimators=100)
classifier_feature.fit(X_train_feat, y_train_all)

RandomForestClassifier()

#### Predicting using the feature extracted model

In [60]:
pred_of_all_features = classifier_all.predict(X_test_all)

In [61]:
pred_of_all_features

array([1, 0, 0, 1, 0, 0, 0, 1, 1, 1, 0, 0, 1, 0, 1, 1, 0, 0, 0, 1, 1, 1,
       0, 0, 1, 1, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 1, 1, 0, 0, 0, 1, 0, 1, 0, 0, 1, 1, 0, 0, 0, 1, 0, 1, 1, 0, 0,
       0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 1, 0, 0, 1, 0, 0, 0, 0, 1,
       1, 0, 0, 0, 1, 0, 0, 0, 1, 0, 1, 0, 1, 1, 0, 0, 0, 0, 1, 1, 0, 0,
       0, 0, 1, 1], dtype=int64)

In [62]:
pre_by_feat = classifier_feature.predict(X_test_feat)

### Finding out the accuracy of the feature extracted model

In [63]:
# accuracy of the model using the whole feature 

print ('Accuracy Score :',accuracy_score(y_test_all, pred_of_all_features))

Accuracy Score : 0.9210526315789473


In [64]:
# accuracy of the model using features from cosal graph

print ('Accuracy Score :',accuracy_score(y_test_feat, pre_by_feat))

Accuracy Score : 0.9385964912280702


**Result:**
- We can observe that the model accuracy using all of the features avaliable is approximatly 92% and and using features extracted from cusal graph is approximally 84%